src/main/java/seedu/address/model/util/SampleDataUtil.java

```diff
@@ -86,28 +109,41 @@ public static User getSampleUser() {
      * @param personData String representing the person data in the text file.
      */
     private static Person getSamplePerson(String personData) {
-        String[] personDataAsArray = personData.split("\\|");
-        Name name = new Name(personDataAsArray[0]);
-        Phone phone = new Phone(personDataAsArray[1]);
-        Email email = new Email(personDataAsArray[2]);
-        Address address = new Address(personDataAsArray[3]);
-        TelegramHandle telegramHandle = new TelegramHandle(personDataAsArray[4]);
-        Set<GroupTag> groupTagSet = getGroupTagSet(personDataAsArray[5].split(" "));
-        Set<ModuleTag> moduleTagSet = getModuleTagSet(personDataAsArray[6].split(" "));
+        List<String> personDataList = Stream.of(personData.split("\\|"))
+                .map(String::trim)
+                .limit(7) // Maximum of 7 fields
+                .collect(Collectors.toList());
+
+        if (personDataList.size() != 7) {
+            return null;
+        }
+
+        Name name = new Name(personDataList.get(0));
+        Phone phone = new Phone(personDataList.get(1));
+        Email email = new Email(personDataList.get(2));
+        Address address = new Address(personDataList.get(3));
+        TelegramHandle telegramHandle = new TelegramHandle(personDataList.get(4));
+        Set<GroupTag> groupTagSet = getGroupTagSetFromUnsplitted(personDataList.get(5));
+        Set<ModuleTag> moduleTagSet = getModuleTagSetFromUnsplitted(personDataList.get(6));
+
```

Comment:
I try to do this to error catch when the sample data is corrupted.

Reply:
LGTM! Suggestion to send a message or throw an error in the future to alert the user or developer that the sample data is corrupted in future milestones.